pkg/utils/tekton/controller.go

```diff
@@ -36,34 +36,35 @@ type KubeController struct {
 }
 
 type Bundles struct {
-	FBCBuilderBundle string
-	DockerBuildBundle string
-	JavaBuilderBundle string
+	FBCBuilderBundle    string
+	DockerBuildBundle   string
+	JavaBuilderBundle   string
 	NodeJSBuilderBundle string
 }
 
-func newBundles(client *kubeCl.K8sClient) (*Bundles, error) {
+func newBundles(client crclient.Client) (*Bundles, error) {
 	namespacedName := types.NamespacedName{
 		Name:      "build-pipeline-selector",
 		Namespace: "build-service",
 	}
 	bundles := &Bundles{}
 	pipelineSelector := &buildservice.BuildPipelineSelector{}
-	err := client.KubeRest().Get(context.TODO(), namespacedName, pipelineSelector)
+	err := client.Get(context.TODO(), namespacedName, pipelineSelector)
 	if err != nil {
 		return nil, err
 	}
 	for _, selector := range pipelineSelector.Spec.Selectors {
 		bundleName := selector.PipelineRef.Name
-		switch (bundleName) {
+		bundleRef := selector.PipelineRef.Bundle
+		switch bundleName {
 		case "docker-build":
-			bundles.DockerBuildBundle = bundleName
+			bundles.DockerBuildBundle = bundleRef
 		case "fbc-builder":
-			bundles.FBCBuilderBundle = bundleName
+			bundles.FBCBuilderBundle = bundleRef
 		case "java-builder":
-			bundles.JavaBuilderBundle = bundleName
+			bundles.JavaBuilderBundle = bundleRef
 		case "nodejs-builder":
-			bundles.NodeJSBuilderBundle = bundleName
+			bundles.NodeJSBuilderBundle = bundleRef
```

Comment:
this is actually the reason why e2e test was failing - bundles struct fields need to point to bundle imageRefs instead of bundle names

Reply:
It is a stupid mistake. Thanks for catching this. :+1: